src/strain.cmd.js

```diff
@@ -343,86 +461,87 @@ class StrainCommand {
+      const strainp = this.setVCL(strainsVCL, 'strains.vcl');
 
-          Promise.all(strainjobs).then(() => {
-            console.log('📕  All dicts have been updated.');
 
-            this.publishVersion(() => {
-              this.purgeAll();
-            });
+      const strainjobs = [];
+      strains.map((strain) => {
+        strainjobs.push(this.putDict('strain_action_roots', strain.name, strain.code).then(() => {
+          console.log(`👾  Set action root for strain  ${strain.name}`);
+        })
+          .catch((e) => {
+            const message = 'Error setting edge dictionary value';
+            console.error(message);
+            console.error(e);
+            throw new Error(message, e);
+          }));
+        strainjobs.push(this.putDict('strain_owners', strain.name, strain.content.owner).then(() => {
+          console.log(`🏢  Set owner for strain        ${strain.name}`);
+        })
+          .catch((e) => {
+            const message = 'Error setting edge dictionary value';
+            console.error(message);
+            console.error(e);
+            throw new Error(message, e);
+          }));
+        strainjobs.push(this.putDict('strain_repos', strain.name, strain.content.repo).then(() => {
+          console.log(`🌳  Set repo for strain         ${strain.name}`);
+        })
+          .catch((e) => {
+            const message = 'Error setting edge dictionary value';
+            console.error(message);
+            console.error(e);
+            throw new Error(message, e);
+          }));
+        if (strain.content.ref) {
+          strainjobs.push(this.putDict('strain_refs', strain.name, strain.content.ref).then(() => {
+            console.log(`🏷  Set ref for strain          ${strain.name}`);
           })
             .catch((e) => {
-              const message = 'Error setting one or more edge dictionary values';
+              const message = 'Error setting edge dictionary value';
               console.error(message);
               console.error(e);
               throw new Error(message, e);
-            });
+            }));
         }
+        if (strain.content.root) {
+          strainjobs.push(this.putDict('strain_root_paths', strain.name, strain.content.root).then(() => {
+            console.log(`🌲  Set content root for strain ${strain.name}`);
+          })
+            .catch((e) => {
+              const message = 'Error setting edge dictionary value';
+              console.error(message);
+              console.error(e);
+              throw new Error(message, e);
+            }));
+        }
+        return strain;
       });
-    });
 
-    return this;
+      // wait for the two initialization jobs to finish, too
+      strainjobs.push(initfp);
+      strainjobs.push(initdp);
+
+      // also wait for the strain.vcl writing to finish
+      strainjobs.push(strainp);
+      strainjobs.push(secretp);
+
+      return Promise.all(strainjobs).then(() => {
```

Comment:
btw `Promise.all()` does not process the jobs sequentially... don't know if this is important.

Reply:
I know, and I'm fine with that.